File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int[] number = Arrays.stream(nums).distinct().toArray();
        int mon = nums.length/2;
        if(number.length >= mon){
            return mon;
        }else{
            return number.length;
        }
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const number = nums.length / 2;
    // 중복 제거
    const pocketmon = new Set(nums);
    // 가져갈 수 있는 포켓몬 수가 포켓몬 종류 수보다 작은 경우
    // 가져갈 수 있는 포켓몬 수 반환
    if (number < Array.from(pocketmon).length){
        return number;
    }
    // 가져갈 수 있는 포켓몬 수가 포켓몬 종류보다 많은 경우
    // 포켓몬 종류 수 반환
    return Array.from(pocketmon).length;
}